custom_components/airscape/fan.py

```diff
@@ -191,7 +201,8 @@ def set_percentage(self, percentage: int):
     def speed_up(self):
         """Instruct fan to increment speed up by 1."""
         try:
-            self._fan.speed_up()
+            if self._speed < self._maximum_speed:
+                self._fan.speed_up()
```

Comment:
This isn't necessary. The driver manages this check. And the fan doesn't care anyway. You can tell it to speed up and it will do nothing if its already at max speed.